### PR TITLE
feat: 11 add new object to application.yml for post request default headers

### PR DESCRIPTION
**Pull request by ewertub**
Add new object to application.yml

### CODE REVIEW

File: src/main/resources/application.yml
Comment (turnerrainer) [inline src/main/resources/application.yml:18]: How is it solved with other request types? GET in the first place but other ones in the future as well.

If it's something that needs finding answers from the code in the future, add a note about it in the docs.

Comment (ewertub) [reply src/main/resources/application.yml:18]: I can add the same thing for GET also and yes it needs a little bit of change in code to add other ones.

File: src/main/java/ee/buerokratt/ruuter/domain/steps/http/HttpQueryArgs.java
Comment (henrikprangelNG) [inline src/main/java/ee/buerokratt/ruuter/domain/steps/http/HttpQueryArgs.java:18]: Something that annoys me a bit, is that the whole ci object is added here. The method would make more sense, if simply accepted a list of Headers that it adds, but currently its executing "business logc" - this isn't really the place to do it. The business logic should be moved to the steps, where that data is needed.

File: src/main/java/ee/buerokratt/ruuter/domain/steps/http/HttpPostStep.java
Comment (henrikprangelNG) [inline src/main/java/ee/buerokratt/ruuter/domain/steps/http/HttpPostStep.java:20]: Considering that i rewrote the httpPost and get methods, then i think it would be reasonable to wait until that logic can me merged into this branch.

**Review by henrikprangelNG — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yml
Comment (turnerrainer) [reply src/main/resources/application.yml:18]: OK

File: src/main/java/ee/buerokratt/ruuter/domain/steps/http/HttpPostStep.java
Comment (turnerrainer) [reply src/main/java/ee/buerokratt/ruuter/domain/steps/http/HttpPostStep.java:20]: Pre-approve by me.

**Review by turnerrainer — APPROVED** [review]

File: src/main/java/ee/buerokratt/ruuter/configuration/ApplicationProperties.java
Comment (henrikprangelNG) [inline src/main/java/ee/buerokratt/ruuter/configuration/ApplicationProperties.java:17]: Instatiate this with `new HttpPost()`

File: src/main/java/ee/buerokratt/ruuter/domain/steps/http/HttpStep.java
Comment (henrikprangelNG) [inline src/main/java/ee/buerokratt/ruuter/domain/steps/http/HttpStep.java:76]: Although it's more "compact" here, i think the try-catch should be in the http helper, considering we use the helper outside of steps aswell, and want it to behave the same way

**Review by henrikprangelNG — CHANGES_REQUESTED** [review]

**Review by henrikprangelNG — APPROVED** [review]

**Pull request merged**
